File: 0704-binary-search/0704-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        index = -1
        l = 0
        r = len(nums) - 1
        
        while l <= r:
            mid = (l + r) // 2
            if nums[mid] > target:
                r = mid - 1
            elif nums[mid] < target:
                l = mid + 1
            else:
                index = mid
                break

        return index
        

File: 0704-binary-search/0704-binary-search.swift
class Solution {
    func search(_ nums: [Int], _ target: Int) -> Int {
        var index = -1
        var l = 0
        var r = nums.count - 1
        var mid = (l + r) / 2
        
        while l <= r {
            if nums[mid] < target {
                l = mid + 1
            } else if nums[mid] > target {
                r = mid - 1
            } else {
                index = mid
                break
            }
            mid = (l + r) / 2
        }
        
        return index
    }
}
